beeline/propagation/honeycomb.py

```diff
@@ -78,9 +91,11 @@ def unmarshal_propagation_context(trace_header):
             parent_id = v
         elif k == 'context':
             context = json.loads(base64.b64decode(v.encode()).decode())
+        elif k == 'dataset':
+            dataset = unquote(v)
```

Comment:
Should we validate dataset has a value before calling `unquote`?

Reply:
I'm pretty sure once we get to here that v will be a string, either empty or containing a dataset. Which either way we are safe to call unquote